torch/_inductor/cudagraph_trees.py

```diff
@@ -1997,6 +2036,13 @@ def shutdown(self):
         self.current_node = None
 
     def record_function(self, new_inputs, function_id) -> List[Optional[Tensor]]:
+        curr_node_id = self._get_node_id()
+        if (
+            self.num_rerecord[curr_node_id][function_id]
+            > torch._inductor.config.triton.cudagraph_max_rerecording_due_to_static_input_idx_mismatch
+        ):
+            return self.ids_to_funcs[function_id].model(new_inputs)
```

Comment:
Do we need both this and the other check above in run ? will we ever hit this path ?